lib/src/services/song_recognition.dart

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2022, Tomás Sasovsky
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:developer';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
+import 'package:nyxx/nyxx.dart';
+import 'package:radio_garden/radio_garden.dart';
+import 'package:uuid/uuid.dart';
+
+class SongRecognitionService {
+  SongRecognitionService._(
+    this.client,
+  ) {
+    _httpClient = http.Client();
+  }
```

Comment:
If we do this instead, we won't have to worry about it being null:

```dart
  SongRecognitionService._(
    this.client,
  ) : _httpClient = http.Client();
```

Reply:
Nice catch!